bundles/org.openhab.io.mqttembeddedbroker/src/main/java/org/openhab/io/mqttembeddedbroker/internal/EmbeddedBrokerService.java

```diff
@@ -303,13 +304,21 @@ public void startEmbeddedServer(@Nullable Integer portParam, boolean secure, @Nu
         // We may provide ACL functionality at some point as well
         IAuthorizatorPolicy authorizer = null;
 
-        if (secure) {
-            server.startServer(new MemoryConfig(properties), null, nettySSLcontextCreator(), authentificator,
-                    authorizer);
-        } else {
-            server.startServer(new MemoryConfig(properties), null, null, authentificator, authorizer);
+        try {
+            if (secure) {
+                server.startServer(new MemoryConfig(properties), null, nettySSLcontextCreator(), authentificator,
+                        authorizer);
+            } else {
+                server.startServer(new MemoryConfig(properties), null, null, authentificator, authorizer);
+            }
+        } catch (IllegalArgumentException e) {
+            if (e.getMessage().contains("Could not deserialize")) {
+                Path persistenceFilePath = Paths.get((new File(persistenceFilename)).getAbsolutePath());
+                logger.warn("persistence corrupt: {}, deleting {}", e.getMessage(), persistenceFilePath);
+                Files.delete(persistenceFilePath);
+                throw  new IllegalStateException("Could not start broker (persistence corrupted)");
```

Comment:
Nice thanks for this patch.
Should it be retried to start the server after the persistence file has been removed?